src/Plugin.cc

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "sdf/Plugin.hh"
+#include "sdf/parser.hh"
+#include "Utils.hh"
+
+using namespace sdf;
+
+class sdf::Plugin::Implementation
+{
+  /// \brief Name of the plugin
+  public: std::string name = "";
+
+  /// \brief Filename of the shared library
+  public: std::string filename = "";
+
+  /// \brief The SDF element pointer used during load.
+  public: sdf::ElementPtr sdf;
+
+  /// \brief SDF elements inside the plugin.
+  public: std::vector<sdf::ElementPtr> contents;
```

Comment:
Do we want to clone these when `Plugin` is copied? I see they are cloned in `Load` and `InsertContent`.

Reply:
Done in 980d5d2